osquery/tables/system/darwin/crash_logs.cpp

```diff
@@ -0,0 +1,146 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <boost/algorithm/string/predicate.hpp>
+#include <boost/algorithm/string/split.hpp>
+#include <boost/algorithm/string/trim.hpp>
+#include <boost/format.hpp>
+#include <boost/regex.hpp>
+
+#include "osquery/tables/system/system_utils.h"
+#include <osquery/core/conversions.h>
+#include <osquery/filesystem.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+namespace fs = boost::filesystem;
+namespace alg = boost::algorithm;
+namespace osquery {
+namespace tables {
+const std::string kDiagnosticReportsPath = "/Library/Logs/DiagnosticReports";
+const std::map<std::string, std::string> kCrashDumpKeys = {
+    {"Process", "pid"},
+    {"Path", "path"},
+    {"Identifier", "identifier"},
+    {"Version", "version"},
+    {"Parent Process", "parent"},
+    {"Responsible", "responsible"},
+    {"User ID", "user_id"},
+    {"Date/Time", "date_time"},
+    {"Crashed Thread", "crashed_thread"},
+    {"Exception Type", "exception_type"},
+    {"Exception Codes", "exception_codes"},
+    {"Exception Note", "exception_notes"},
+    {"rax", "rax"},
+    {"rbx", "rbx"},
+    {"rcx", "rcx"},
+    {"rdx", "rdx"},
+    {"rdi", "rdi"},
+    {"rsi", "rsi"},
+    {"rbp", "rbp"},
+    {"rsp", "rsp"},
+};
+
+void readCrashDump(const std::string& app_log, Row& r) {
+  r["log_location"] = app_log;
+  std::string content;
+
+  if (readFile(app_log, content).ok()) {
+
+    // Variables for capturing the stack trace
+    boost::format crashed_thread_format("Thread %1% Crashed");
+    auto crashed_thread_seen = false;
+
+    auto lines = split(content, "\n");
+    for (auto it = lines.begin(); it != lines.end(); it++) {
+      auto line = *it;
+      // Tokenize first by colons
+      auto toks = split(line, ":");
+
+      if (toks.size()) {
```

Comment:
This is an integer comparison that's been implicitly coerced into a boolean statement. Both the code and my last sentence are confusing! Try:
```cpp
if (toks.size() == 0) {
  continue;
}
```